src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -127,6 +146,23 @@ public void init(Vertx vertx, Context context) {
         }
     }
 
+    /**
+     * Get the metric configuration.
+     *
+     * @param metricsConfig the global {@link MetricsConfig} object.
+     * @return the metric configuration as JSON object.
+     */
+    @VisibleForTesting
+    JsonObject getMetricsConfig(MetricsConfig metricsConfig) {
+        JsonObject metrics = config().getJsonObject(CONFIG_METRICS_KEY, new JsonObject());
+        Boolean enabled = metrics.getBoolean(ConfiguredDataVerticleMetrics.ENABLED);
+        // if there is no setting for the metric enabled in the DataVerticle configuration use the global setting.
+        if (enabled == null || Boolean.TRUE.equals(enabled)) {
+            metrics.put(ConfiguredDataVerticleMetrics.ENABLED, metricsConfig.isEnabled());
+        }
```

Comment:
```suggestion
        JsonObject metrics = config().getJsonObject(CONFIG_METRICS_KEY, new JsonObject());
        boolean enabled = metrics.getBoolean(ConfiguredDataVerticleMetrics.ENABLED, true);
        // if there is no setting for the metric enabled in the DataVerticle configuration use the global setting.
        if (enabled) {
            metrics.put(ConfiguredDataVerticleMetrics.ENABLED, metricsConfig.isEnabled());
        }
```

Reply:
In your case I would get a NullPointerExcpetion if the ConfiguredDataVerticleMetrics.ENABLED key has the value null.